src/services/connextService.ts

```diff
@@ -0,0 +1,96 @@
+import { getPoolFeeForUniV3, getXCallCallData, prepareSwapAndXCall } from '@connext/chain-abstraction';
+import { DestinationCallDataParams, Swapper, SwapAndXCallParams } from '@connext/chain-abstraction/dist/types';
+import { SdkConfig, create } from '@connext/sdk';
+import { RAW_NETWORKS, SUPPORTED_CHAINS_BY_CONNEXT } from 'config';
+import WalletService from './walletService';
+
+interface DomainID {
+  [key: number]: string;
+}
+
+export default class ConnextService {
+  sdkConfig: SdkConfig;
+
+  walletService: WalletService;
+
+  destinantionChainID: number;
+
+  constructor(walletService: WalletService, destinantionChainID: number) {
+    this.walletService = walletService;
+    this.destinantionChainID = destinantionChainID;
+  }
+
+  getRPCURL(chainID: number) {
+    const network = Object.values(RAW_NETWORKS).find((net) => net.chainId === chainID);
+    return network?.rpc[0];
+  }
+
+  async getCalculatedRelayerFees(originDomain: string, destinationDomain: string) {
+    const domainConfig: { [domainId: string]: { providers: string[] } } = {};
+
+    const domainChainIds = Object.entries(SUPPORTED_CHAINS_BY_CONNEXT)
+      .filter(([key]) => typeof key === 'number')
+      .map(([key, value]) => ({ domainId: value.domainId, chainId: key }));
+
+    domainChainIds.forEach((obj) => {
+      domainConfig[obj.domainId] = { providers: [this.getRPCURL(parseInt(obj.chainId, 10)) as string] };
```

Comment:
obj.chainId is already a string, no need to parse it

Reply:
we should check that `this.getRPCUrl` actually does return something in case it could find the chainId